### PR TITLE
Fix tests in master

### PR DESCRIPTION
**Pull request by RikkiGibson**
Interaction between #46441 and #46616 caused some tests to break. Fixing them up now.

### CODE REVIEW

**Comment by jcouv** [comment]
>     public Error field;

I didn't understand why we complain on this field. It could be assigned and read by other parts of the program.

---
Refers to: src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs:4242 in 8a2d8e5. [](commit_id = 8a2d8e5e0ccaf0baab9844a1b111d12e554ba745, deletion_comment = False)

File: src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs [hunk 1]
```diff
@@ -4272,7 +4275,23 @@ record C1
...
+                // (12,22): warning CS0649: Field 'C1.field5' is never assigned to, and will always have its default value null
+                //     protected string field5;
+                Diagnostic(ErrorCode.WRN_UnassignedInternalField, "field5").WithArguments("C1.field5", "null").WithLocation(12, 22),
+                // (13,31): warning CS0649: Field 'C1.field6' is never assigned to, and will always have its default value null
+                //     protected internal string field6;
+                Diagnostic(ErrorCode.WRN_UnassignedInternalField, "field6").WithArguments("C1.field6", "null").WithLocation(13, 31),
+                // (14,30): warning CS0649: Field 'C1.field7' is never assigned to, and will always have its default value null
+                //     private protected string field7;
+                Diagnostic(ErrorCode.WRN_UnassignedInternalField, "field7").WithArguments("C1.field7", "null").WithLocation(14, 30)
+);
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs:4294]: nit: formatting

Comment (RikkiGibson) [reply src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs:4294]: oops! :(

**Comment by jcouv** [comment]
>     internal int Property3;

Same feedback for Property3 here, since it is accessible to the rest of the program.

---
Refers to: src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs:4347 in 8a2d8e5. [](commit_id = 8a2d8e5e0ccaf0baab9844a1b111d12e554ba745, deletion_comment = False)

**Comment by RikkiGibson** [comment]
> I didn't understand why we complain on this field. It could be assigned and read by other parts of the program.

My read of it is that the field has an effective accessibility of 'internal', thus we "know" whether or not the field was written or read. Compare to the [analogous scenario](https://sharplab.io/#v2:EYLgtghgzgLgpgJwD4AEBMACAwgRgLABQA3oRmRigMwYCiCCA9ghgGYCWcANgCYDchAXyA==) involving a class instead of a record.

**Comment by jcouv** [comment]
From discussion with Rikki, I learned that we do analysis across types for internal fields. I didn't realize that.
So adding a usage somewhere else resolves the warning.
Good to go then!

```
        [Fact]
        public void ToString_TopLevelRecord_OneField_ErrorType()
        {
            var src = @"
record C1
{
    public Error field;
}
class D
{
    void M(C1 c)
    {
        c.field = null;
    }
}
";

            var comp = CreateCompilation(src);
            comp.VerifyEmitDiagnostics(
                // (4,12): error CS0246: The type or namespace name 'Error' could not be found (are you missing a using directive or an assembly reference?)
                //     public Error field;
                Diagnostic(ErrorCode.ERR_SingleTypeNameNotFound, "Error").WithArguments("Error").WithLocation(4, 12)
                );
        }
```

**Comment by sharwell** [comment]
> From discussion with Rikki, I learned that we do analysis across types for internal fields.

This analysis is disabled as soon as the project contains an IVT, even if it's just an IVT to tests. In practice that hides the diagnostics for most of our projects.

**Comment by RikkiGibson** [comment]
I have successfully built with analyzers and run tests locally, and I want to unblock the master build, so I will just merge now.

**Pull request merged**